src/execution_plan.c

```diff
@@ -1694,10 +1755,72 @@ int ExecutionPlan_ExecutionsDump(RedisModuleCtx *ctx, RedisModuleString **argv,
 	return REDISMODULE_OK;
 }
 
+int ExecutionPlan_ExecutionGet(RedisModuleCtx *ctx, RedisModuleString **argv, int argc){
+	if(argc != 2){
+		return RedisModule_WrongArity(ctx);
+	}
+
+	const char* id = RedisModule_StringPtrLen(argv[1], NULL);
+	ExecutionPlan* ep = RedisGears_GetExecution(id);
+
+	if(!ep){
+		RedisModule_ReplyWithError(ctx, "execution plan does not exist");
+		return REDISMODULE_OK;
+	}
+
+    pthread_mutex_lock(&epData.mutex);
```

Comment:
why you need to acquire this lock?

Reply:
Hmmm - I assumed it is needed for the same reason it exists in DUMPEXECUTIONS. Is that not the case?

Reply:
Sorry you are right, now the problem that between the time you took the execution to the time locking it might be that the execution got deleted. We are gone need a lock that allow the same thread to lock it multiple times (like I did with the LockHandler). Let me sleep on a good solution for that..

Reply:
This has been "resolved" with a TODO and a big warning